lib/src/commit_builder.rs

```diff
@@ -85,6 +85,15 @@ impl CommitBuilder<'_> {
         {
             commit.author.email = commit.committer.email.clone();
         }
+
+        // Reset timestamp on discardable commits if author is committer
+        if commit.author.name == commit.committer.name
+            && commit.author.email == commit.committer.email
+            && predecessor.is_discardable(mut_repo).unwrap_or_default()
```

Comment:
nit: we could optimize this by taking merge commit as non-resettable, but I'm not sure if it's worth the effort.

Reply:
I would prefer not to treat merge commits differently. That would mean that if you start working on a commit that's based on multiple other commits (i.e. it's a merge commit), then the timestamp would not get updated on that commit when it gets rewritten. That would seem surprising to me.